exoscale/api/storage.py

```diff
@@ -343,6 +343,21 @@ def metadata(self):
 
         return res["Metadata"]
 
+    @property
+    def url(self):
+        """
+        Stored file URL.
+
+        Returns:
+            str: the stored file URL
+
+        Note:
+            This property value is dynamically retrieved from the API, incurring extra
+            latency.
+        """
+
+        return join(self.bucket.url, self.path)
```

Comment:
Using `os.path.join` for building urls is not a good idea as it uses `os.sep` , you'll get different results depending on your OS. 

You should do something like


```
        p = self.path.replace(os.sep, "/")
        return "/".join([self.bucket.url, p])
```

Reply:
Good catch, thanks @jgrondier! I've changed for `urllib.parse.urljoin()` instead: 44339bc

Reply:
It still wouldn't work as expected on filesystems with a different path separator.

```
In [10]: url = "https://example.com"                                                                                                                                                

In [11]: windows_path = "var\img"                                                                                                                                                  

In [12]: urljoin(url, windows_path)                                                                                                                                                
Out[12]: 'https://example.com/var\\img'
```

I unfortunately don't have a Windows machine to test interactions, but `pathlib` module might have a solution for that. I don't think I'm seeing something to check that the path is relative either (If it even matters) ?. But I might be mistaken.

Reply:
Well after reading the `urllib.parse.urljoin()` it doesn't actually does what I want, so finally ended up with a good ol' manual slash stripping: 95b5412